### PR TITLE
Fall back to Jest defaults where possible

### PR DESCRIPTION
**Pull request by jonkoops**
Cleans up the Jest configuration file as much as possible by falling back to the default values where possible and where it makes sense. This PR also removes the `jest-junit` reporter, as it was not being used anywhere.

### CODE REVIEW

File: jest.config.ts [hunk 1]
```diff
@@ -5,11 +5,6 @@ const jestConfig: JestConfigWithTsJest = {
   moduleNameMapper: {
     "^(\\.{1,2}/.*)\\.js$": "$1",
   },
-  testMatch: ["**/test/**/*.test.ts", "**/test/tests.ts"],
-  coverageProvider: "v8",
```
Comment (frederikprijck) [inline jest.config.ts:9]: Can you elaborate on why you want this to be changed and why we should? We set this to V8, but this would use babel again.

Comment (jonkoops) [reply jest.config.ts:9]: I checked the documentation and some blog posts, but I could not find any distinct advantages deviating from the default here. If there is a good reason to keep it I see no problem restoring this, WDYT?

Comment (frederikprijck) [reply jest.config.ts:9]: v8 would use native coverage in node, which I tend to prefer and use in most of the libraries I maintain. I don't feel super strong about this, but I also feel like we shouldn't just change things only for the sake of changing them.

File: jest.config.ts [hunk 2]
```diff
@@ -5,11 +5,6 @@ const jestConfig: JestConfigWithTsJest = {
...
-  coveragePathIgnorePatterns: ["/node_modules/", "./test"],
-  reporters: ["default", ["jest-junit", { outputDirectory: "test-results/jest" }]],
-  coverageReporters: ["lcov", "text", "text-summary"],
```
Comment (frederikprijck) [inline jest.config.ts:12]: Let's bring back the coverageReporters, as this changes the coverage output in the terminal.

**Review by frederikprijck — CHANGES_REQUESTED** [review]

File: jest.config.ts (hunk 2 above)
Comment (jonkoops) [reply jest.config.ts:12]: Done 👍

**Review by frederikprijck — APPROVED** [review]

**Pull request merged**